lib/std/os.zig

```diff
@@ -1543,29 +1551,36 @@ pub const SymLinkError = error{
 /// one; the latter case is known as a dangling link.
 /// If `sym_link_path` exists, it will not be overwritten.
 /// See also `symlinkC` and `symlinkW`.
-pub fn symlink(target_path: []const u8, sym_link_path: []const u8) SymLinkError!void {
+pub fn symlink(target_path: []const u8, sym_link_path: []const u8, flags: SymlinkFlags) SymLinkError!void {
```

Comment:
Consider that it is planned to rename `std.os` to `std.posix` (see #5019). It follows then that this `symlink` function should not have this flags parameter, and if it is not possible to implement a posix-compliant symlink function on Windows, then it should be a compile error for that target.

Wrappers for Windows DLL calls should go into `std.os.windows` and then the cross-platform abstractions (e.g. `std.fs`) can call either the posix one or the windows one depending on OS. The cross-platform abstractions is the proper place to have this flags parameter.

Reply:
Makes sense, and thanks for the link to the issue!

Reply:
While we're here, should we then make it a compile error when trying to call `std.os.symlink` or should we always create a file symlink on windows?

Reply:
I think we should either implement the posix behavior, or make it a compile error. So for now, a compile error would be best. I'm sure some time in the future someone is going to want to use the zig std lib to make a windows libc implementation, and implement a posix layer for this API, but it's not a priority for me personally.